src/librustc_mir/borrow_check/nll/region_infer.rs

```diff
@@ -54,6 +56,11 @@ pub struct RegionInferenceContext<'tcx> {
 
     num_free_regions: usize,
 
+    /// The map explains what caused some region `R` with index
+    /// `RegionVid` to be related to some `RegionElement` `RE`.
+    causes: IndexVec<RegionVid, HashMap<RegionElement, Cause>>,
```

Comment:
👍

Reply:
Although, I think this would work out more cleanly as a `FxHashMap<(RegionVid, RegionElement), Cause>`, to be honest. I'm just going to pretend it was that. =)